src/software/proto/primitive/primitive_msg_factory.h

```diff
@@ -42,28 +64,14 @@ MAKE_ENUM(MaxAllowedSpeedMode, PHYSICAL_LIMIT, STOP_COMMAND);
  * @param dribbler_mode The dribbler mode
  * @param auto_chip_or_kick The command to autochip or autokick
  * @param max_allowed_speed_mode The mode of maximum speed allowed
+ * @param target_spin_rps The target spin while moving in revolutions per second
  *
  * @return Pointer to Move Primitive Message
  */
 std::unique_ptr<TbotsProto::Primitive> createMovePrimitive(
     const Point &dest, double final_speed_m_per_s, const Angle &final_angle,
     DribblerMode dribbler_mode, AutoChipOrKick auto_chip_or_kick,
-    MaxAllowedSpeedMode max_allowed_speed_mode);
-
-/**
- * Create a Spinning Move Primitive Message
- *
- * @param dest The final destination of the movement
- * @param final_speed_meters_per_second The speed at final destination
- * @param angular_velocity The angular velocity of the robot
- * of the movement
- * @param dribbler_mode The dribbler mode
- *
- * @return Pointer to Spinning Move Primitive Message
- */
-std::unique_ptr<TbotsProto::Primitive> createSpinningMovePrimitive(
-    const Point &dest, double final_speed_meters_per_second,
-    const AngularVelocity &angular_velocity, DribblerMode dribbler_mode);
+    MaxAllowedSpeedMode max_allowed_speed_mode, double target_spin_rps);
```

Comment:
Seeing how much we instantiate `target_spin_rps` to be 0 in the tests, would it be better to make `target_spin_rps` default to 0?

Reply:
No, it's better to explicitly set values than to rely on default values, because of potential misunderstandings, i.e. what is the right default value? The factory could be used in so many ways. I've added default values at the tactic level because there's enough context from how we expect the move tactic to work to have default values.